search/intermediate_result.cpp

```diff
@@ -281,10 +281,12 @@ void FillDetails(FeatureType & ft, Result::Details & details)
     }
   }
 
-  if (strings::to_uint(ft.GetMetadata(feature::Metadata::FMD_STARS), details.m_stars))
+  details.m_isHotel = feature::IsHotel(ft);
+  if (details.m_isHotel &&
+      strings::to_uint(ft.GetMetadata(feature::Metadata::FMD_STARS), details.m_stars))
```

Comment:
Is IsHotel check necessary here? I suspect that it's already checked in generator. Or why not to push stars wherever they are defined in OSM ..
Anyway, I think we can skip core hpp/cpp changes for this patch.

Reply:
I added `details.m_isHotel = feature::IsHotel(ft);` line because this field presents in struct but not filled.
I don't know how fast `strings::to_uint()` and  `ft.GetMetadata()` works, but I think it will be much faster to check `m_isHotel` first
And `if (details.m_isHotel)` check was added because there is a comment in code that `m_stars` used only for hotels: [link](https://github.com/organicmaps/organicmaps/blob/60b1e5e41eccb9fac8227ff9539e51f586edbc12/search/result.hpp#L59)

So, you think I should revert changes in this file?

Reply:
Hm, I didn't notice that we have uninitialized details.m_isHotel var, thanks.
Ok then. Please, keep
```
else
  details.m_stars = 0;
```

Reply:
And please test search on map for hotels. There should be a valid Hotel icon after your changes instead of a regular blue dot.

Reply:
Done.